### PR TITLE
API spec-compliance fixes

### PR DESCRIPTION
**Pull request by atolopko-czi**
Resolves https://github.com/single-cell-data/TileDB-SOMA/issues/834
Resolves https://github.com/single-cell-data/SOMA/issues/133

### CODE REVIEW

File: abstract_specification.md [hunk 1]
```diff
@@ -940,18 +938,15 @@ To facilitate distributed computation, read operations on foundational types acc
...
-| `csr`          | Return an `Arrow.SparseCSRTensor`                                                                      |
-| `csc`          | Return an `Arrow.SparseCSCTensor`                                                                      |
-| `record-batch` | Return an `Arrow.RecordBatch` containing COO-encoded coordinates and values.                           |
-| `table`        | Return an `Arrow.Table` containing COO-encoded coordinates and values.                                 |
+| Format         | Description                                                                                   |
+|----------------|-----------------------------------------------------------------------------------------------|
+| `dense`        | Return an iterator of `Arrow Tensor`s containing slice values.                                |
+| `coos`         | Return an iterator of `Arrow.SparseCOOTensor`s containing COO-encoded coordinates and values. |
+| `table`        | Return an iterator of `Arrow.Table`s containing COO-encoded coordinates and values.           |
```
Comment (atolopko-czi) [inline abstract_specification.md:949]: Not sure if csr, csc, and record-batch should be removed from the spec, but as they're no longer in the tiledbsoma impl, I removed them for now. If intended to remain, perhaps we should indicate that implementations can optionally implement formats.

Comment (johnkerl) [reply abstract_specification.md:949]: They need to be removed everywhere & it's a mistake on my part if I missed removing them from anywhere.

Comment (atolopko-czi) [reply abstract_specification.md:949]: Thanks for confirming. Will stick with the change as is; no other occurrences exist.

**Review by johnkerl — APPROVED** [review]

File: abstract_specification.md [hunk 2]
```diff
@@ -1190,3 +1185,7 @@ Issues to be resolved:
 44. Pull description of common operations into its own section.
 45. Specify object lifecycle and related operations (`create`, `open`, `add_new_*`, etc.).
 46. Uniformize backticks, punctuation, etc.
+47. Fixed erroneous backticks, spelling, and capitalization. 
+48. Renamed `SOMADataFrame.read()` `ids` param and `SOMASparseNDArray.read()` `slice` param to `coords`, for consistency between `SOMADataFrame`, `SOMASparseNDArray` and `SOMADenseNDArray` types. 
+49. Updated `SOMABatchFormat` section, renaming to `SOMASparseNDArrayRead` and removing the `csr`, `csc`, and `record-batch` format options.
+50. Removed `SOMASparseNDArray.read()` `batch_format` param and changed return type to `SOMASparseNDArrayRead`.
```
Comment (atolopko-czi) [inline abstract_specification.md:1191]: - [ ] Will also need to update the 2 occurrences of this text in the `get schema` descriptions of the `*NDArray` sections: "This operation will return the schema of the `Arrow.RecordBatch` returned
by the `read` operation when it is called with a `batch_format` parameter value of `record-batch`."

File: abstract_specification.md (hunk 1 above)
Comment (atolopko-czi) [reply abstract_specification.md:950]: Actually, `record-batch` is still in the core code so reinstated here

**Comment by atolopko-czi** [comment]
> As above

Sorry, I'm not following what change is being requested by this.

**Comment by bkmartinjr** [comment]
changes to date seem good.  I left a few suggestions in single-cell-data/TileDB-SOMA#834 about additional cleanup of the open `issues` or `to be further specified` notes.

File: abstract_specification.md [hunk 3]
```diff
@@ -243,16 +241,14 @@ Composed types are defined as a composition of foundational types, adding name,
...
-- `SOMAMeasurement`: for all observables, a common set of annotated variables (defined by a `SOMADataFrame`) for which values (e.g., measurements, calculations) are stored in `SOMADenseNdArray` and `SOMASparseNdArray`.
+- `SOMAMeasurement`: for all observables, a common set of annotated variables (defined by a `SOMADataFrame`) for which values (e.g., measurements, calculations) are stored in `SOMADenseNDArray` and `SOMASparseNDArray`.
 
 In other words, every `SOMAMeasurement` has a distinct set of variables (features), and inherits common observables from its parent `SOMAExperiment`. The `obs` and `var` dataframes define the axis annotations, and their respective `soma_joinid` values are the indices for all matrixes stored in the `SOMAMeasurement`.
 
 <figure>
     <img src="images/SOMAExperiment.png" alt="SOMAExperiment">
 </figure>
 
-> ⚠️ **Issue**: it would be a good idea to factor `SOMAExperiment` and `SOMAMeasurement` into separate sections.
```
Comment (atolopko-czi) [inline abstract_specification.md:254]: Now an issue: https://github.com/single-cell-data/SOMA/issues/147

File: abstract_specification.md [hunk 4]
```diff
@@ -1131,17 +1111,6 @@ The format and contents of a Context object is completely implementation-defined
 An implementation may provide a way to construct its specific context type, which can be used by user setup code to connect to the SOMA data store (represented above as the `somaimpl.create_context` call).
 However, client code should treat the `context` object on any instantiated SOMA objects as an opaque value, and only pass it directly as the context parameter when creating or opening other SOMA data.
 
-# ⚠️ Other Issues (open issues with this doc)
-
-Issues to be resolved:
-
-1. Are there operations specific to SOMAExperiment and SOMAMeasurement that need to be defined? Or do they inherit only the ops from SOMACollection?
-2. What (if any) additional semantics around writes need to be defined?
-3. Value filter support in `NDArray` has been proposed:
```
Comment (atolopko-czi) [inline abstract_specification.md:1140]: Moved to https://github.com/single-cell-data/SOMA/issues/148

File: abstract_specification.md [hunk 5]
```diff
@@ -1131,17 +1111,6 @@ The format and contents of a Context object is completely implementation-defined
 An implementation may provide a way to construct its specific context type, which can be used by user setup code to connect to the SOMA data store (represented above as the `somaimpl.create_context` call).
 However, client code should treat the `context` object on any instantiated SOMA objects as an opaque value, and only pass it directly as the context parameter when creating or opening other SOMA data.
 
-# ⚠️ Other Issues (open issues with this doc)
-
-Issues to be resolved:
-
-1. Are there operations specific to SOMAExperiment and SOMAMeasurement that need to be defined? Or do they inherit only the ops from SOMACollection?
-2. What (if any) additional semantics around writes need to be defined?
```
Comment (atolopko-czi) [inline abstract_specification.md:1139]: Deleting #2 for good. I believe write semantics have been well documented now in the `Object Lifecycle` section, w.r.t. atomicity and concurrency semantics.

File: abstract_specification.md [hunk 6]
```diff
@@ -1131,17 +1111,6 @@ The format and contents of a Context object is completely implementation-defined
 An implementation may provide a way to construct its specific context type, which can be used by user setup code to connect to the SOMA data store (represented above as the `somaimpl.create_context` call).
 However, client code should treat the `context` object on any instantiated SOMA objects as an opaque value, and only pass it directly as the context parameter when creating or opening other SOMA data.
 
-# ⚠️ Other Issues (open issues with this doc)
-
-Issues to be resolved:
-
-1. Are there operations specific to SOMAExperiment and SOMAMeasurement that need to be defined? Or do they inherit only the ops from SOMACollection?
```
Comment (atolopko-czi) [inline abstract_specification.md:1138]: Bruce votes for deleting and so do I, so deleting (1) for good.

File: abstract_specification.md [hunk 7]
```diff
@@ -70,6 +70,7 @@ The foundational types are:
 The composed types are:
 
 - `SOMAExperiment`: a specialization and extension of `SOMACollection`, codifying a set of naming and indexing conventions to represent an annotated, 2-D matrix of observations across _multiple_ sets of variables.
+- `SOMAMeasurement`: a specialization and extension of `SOMACollection`, that contains a set of annotated observables that are common to one or more sets of measurements and/or derived calculations.
```
Comment (atolopko-czi) [inline abstract_specification.md:73]: For https://github.com/single-cell-data/SOMA/issues/133

**Comment by atolopko-czi** [comment]
@thetorpedodog @mlin Let me know if you intend to review these spec changes, in which case I'll hold off on merging. Ty!

**Review by thetorpedodog — APPROVED** [review]
